receiver.go

```diff
@@ -542,7 +541,7 @@ func (r *Receiver) attach(ctx context.Context) error {
 }
 
 func (r *Receiver) mux() {
-	defer r.l.muxDetach(context.Background(), func() {
+	defer r.l.muxClose(context.Background(), r.detachError, func() {
```

Comment:
This'll bake in r.detachError, correct? Instead of getting the value of r.detachError when the mux() ends?

Reply:
Hmm yes you're correct.  This is weird though because we have a test for peer-initiated detach and it's still passing.  Digging deeper....

Reply:
The test verified the values on the client side but not the server side.  I've fixed this and updated the tests.  Great catch!